src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1079,7 +1079,7 @@ public Content seeTagToContent(Element element, DocTree see, TagletWriterImpl.Co
         } else if (refMemName == null) {
             // Must be a class reference since refClass is not null and refMemName is null.
             if (labelContent.isEmpty()) {
-                if (!refClass.getTypeParameters().isEmpty() && seeText.contains("<")) {
+                if (seeText.contains("<")) {
```

Comment:
Hmmm, string-based checks are generally something of a red-flag, as compared to more abstract methods on appropriate classes. And, this is a string operation on the output of  `CommentHelper.getText`. Is there a way that we have may this a more robust test ... for example introduce `CommentHelper.isGeneric(ReferenceTree)` or something like that?

Reply:
I agree something like `CommentHelper.isGeneric(ReferenceTree)` would be preferable, but that method would have to iterate over enclosing types to find any generic types there, just like the `LinkInfo.linkEnclosingTypes(DeclaredType)` method I have added. So I wonder where a shared method for this could go if not to `Utils` (I shudder to think).

Reply:
As much as I dislike `Utils` it is probably the correct location in this case.  I think of `Utils` as being javadoc-specific enhancements and variations on the Language Model `Elements` and `Types` interfaces.